lib/src/main/scala/spinal/lib/bus/amba4/axi/Axi4Unburster.scala

```diff
@@ -0,0 +1,295 @@
+          respOut := resp
+
+          io.retIdResp.ready := valid
+
+          when(io.retIdResp.fire) {
+            addrBeats := addrBeats - 1
+            when(!resp.orR) {
+              resp := io.retIdResp.resp
+              respOut := io.retIdResp.resp
+            }
+          }
+
+          valid.clearWhen(io.retIdResp.fire && addrBeats === 0)
+          io.last := addrBeats === 0
+        }
+      }
+    }
+
+  } else {
+    // Length FIFO
+    val fifo = new StreamFifo(config.lenType.clone, pendingDepth)
+    val lenInStream = fifo.io.push
+    val lenFifo = fifo.io.pop
+
+    // Working length
+    val valid = RegInit(False)
+    val addrBeats = Reg(UInt(8 bit)) init(0)
+    val resp = Reg(Bits(2 bit)) init(0)
+
+    // Fork into the length FIFO
+    io.axIdLen.ready := lenInStream.ready
+    lenInStream.valid := io.axIdLen.valid
+    lenInStream.payload := io.axIdLen.len
+
+    lenFifo.ready := !valid
+
+    when(lenFifo.fire) {
+      addrBeats := lenFifo.payload
+      resp.clearAll()
+      valid.set()
+    }
+
+    when(io.retIdResp.fire) {
+      addrBeats := addrBeats - 1
+      when(!resp.orR) {
+        resp := io.retIdResp.resp
+      }
+    }
+
+    valid.clearWhen(io.retIdResp.fire && addrBeats === 0)
+    io.retIdResp.ready := valid
+    io.last := addrBeats === 0
+    io.resp := resp
+
+//    io.drop := !(valid || fifo.io.occupancy.orR)
+  }
+}
+
+/**
+  * Converts Axi4 burst streams into single beat transactions and adds last as required.
+  * AR channel will block if the pending transactions FIFO is full for that ID.
+  *
+  * @param config Axi4Config of the inbound master stream
+  * @param pendingDepth Number of pending transactions per ID
+  * @param pendingWidth Number of concurrent pending ID transactions.
+  *                     Not used if the master stream does not support IDs.
+  */
+class Axi4ReadOnlyUnburster(config: Axi4Config, pendingDepth: Int = 3, pendingWidth: Int = 3) extends Component {
+  val io = new Bundle {
+    val input = slave(Axi4ReadOnly(config))
+    val output = master(Axi4ReadOnly(config.copy(useLen = false, useBurst = false)))
+  }
+
+  if (!config.useLast) {
+    val unburstified = io.input.ar.unburstify
+
+    io.output.ar.arbitrationFrom(unburstified)
+    io.output.ar.payload.assignSomeByName(unburstified.fragment)
+
+    io.input.r.arbitrationFrom(io.output.r)
+    io.input.r.payload.assignSomeByName(io.output.r.payload)
+  } else {
+
+    val manager = new UnbursterIDManager(config, pendingDepth, pendingWidth)
+
+    val unburstified = io.input.ar.continueWhen(manager.io.axIdLen.ready).unburstify
+    manager.io.axIdLen.valid := io.input.ar.fire
+    (config.useId) generate { manager.io.axIdLen.id := io.input.ar.id }
+    manager.io.axIdLen.len := io.input.ar.len
+
+    io.output.ar.arbitrationFrom(unburstified)
+    io.output.ar.payload.assignSomeByName(unburstified.fragment)
+
+    val rFifo = io.output.r.queueLowLatency(8).continueWhen(manager.io.retIdResp.ready)
```

Comment:
I think that fifo shouldn't be integrated in the bridge, to keep things lite, and potentialy the user can added it externaly, or maybe via a parameter.
Mainly, if you use queueLowLatency, on FPGA without distributed ram, it will be very costly, as it will be flip flop based